### DISCUSSION TITLE
how to get ProtoBuf file from other Valve games such as HL、l4d2、protal2

### DISCUSSION THREAD

**Discussion by ghost** [SteamRE/SteamKit#1045] [General]
Hello there,how to get ProtoBuf file from other Valve games such as HL、l4d2、protal2

**Comment by psychonic** [comment]
Half-Life 2 and Left 4 Dead 2 don't use protobuf for anything. The Portal 2 GC messages can be dumped with [ProtobufDumper](https://github.com/SteamRE/SteamKit/tree/master/Resources/ProtobufDumper).
